File: daily/452.用最少数量的箭引爆气球.js
/*
 * @lc app=leetcode.cn id=452 lang=javascript
 *
 * [452] 用最少数量的箭引爆气球
 */

// @lc code=start
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
  let count = 0
  points.sort((a, b) => a[1] - b[1])
  for (let i = 0; i < points.length; ++i) {
    const p = points[i]
    while (i + 1 < points.length && points[i+1][0] <= p[1]) i++
    count++
  }
  return count
};
// @lc code=end

const points = [[1,2],[2,3],[3,4],[4,5]]
console.log(findMinArrowShots(points))

File: daily/452.用最少数量的箭引爆气球.rs
/*
 * @lc app=leetcode.cn id=452 lang=rust
 *
 * [452] 用最少数量的箭引爆气球
 */

// @lc code=start
impl Solution {
    pub fn find_min_arrow_shots(points: Vec<Vec<i32>>) -> i32 {
        let mut i = 0;
        let mut count = 0;
        let mut points = points;
        points.sort_by_cached_key(|x| x[1]);
        while i < points.len() {
            let end = points[i][1];
            while i + 1 < points.len() && points[i + 1][0] <= end { i += 1; }
            i += 1;
            count += 1;
        }
        return count;
    }
}
// @lc code=end

